modules/python/src2/cv2.cpp

```diff
@@ -343,10 +344,21 @@ static bool pyopencv_to(PyObject* o, Mat& m, const ArgInfo info)
         _strides = PyArray_STRIDES(oarr);
     }
 
-    for(int i = 0; i < ndims; i++)
+    // Normalize strides in case NPY_RELAXED_STRIDES is set
+    size_t default_step = elemsize;
+    for ( int i = ndims - 1; i >= 0; --i )
     {
         size[i] = (int)_sizes[i];
-        step[i] = (size_t)_strides[i];
+        if ( size[i] > 1 )
+        {
```

Comment:
Side Remark: I personally always think, if you know that your arrays are contiguous, just don't use stride information at all. To choke on the mangled up strides, the function has to use the strides, while at the same time assuming contiguity, either one is fine, but both is not great habit IMO.